site/en/education/recommendations-for-web-views.md

```diff
@@ -0,0 +1,48 @@
+---
+title: Recommendations on using Web Views
+metadesc: Recommendations for developers to stay compliant with content filtering requirements that schools may have when using embedded webviews.
+date: 2022-04-19
+weight: -3
+tags:
+  - web
+---
+
+Education institutions use content filtering products to prevent their users from accessing inappropriate content on their Chromebooks. There are a handful of ways that their users are able to escape this content sandbox and access content that they should otherwise not be able to access. We have compiled a list of recommendations for Android and Chrome App developers targeting the education market that should help with making sure that your applications work well for education users.
+
+## Understanding WebViews in school scenarios
+
+If your app uses WebViews to show content such as to allow students to login to your service or to display documentation, it may enable students to browse the web without the content filtering protections provided by their schools. Some of the most popular content filtering products used by schools on Chromebooks are deployed as Chrome extensions, which can only see traffic within browser tabs. Since web content is often unpredictable, students can often traverse links within the WebView to browse the web freely, without content filtering protections. Carefully evaluate the use of WebViews in your app and consider the alternatives below.
+
+## General App Development Advice
+
+### Reduce or remove use of web browser content
+
+By reducing the amount of browser based content shown in your application, you can reduce the likelihood of unwanted content being accessible within your application. If this is not possible, then ensure your WebView only serves content you want and does not link out to a search engine or the open web.
+
+### Ensure your WebView only serves content you want and does not link out
+
+If your app requires that a webview is within your application, make sure that the content within your webview is not able to escape your application and navigate to content that you do not own or control. This can be done by modifying the [webview client to override loading certain URLs via an allowlist](<https://developer.android.com/reference/android/webkit/WebViewClient#shouldOverrideUrlLoading(android.webkit.WebView,%20android.webkit.WebResourceRequest)>) on Android or by manually parsing through your site and validating that the links do not link out to third party content that you cannot control as this may enable your users to go to inappropriate content. For Chrome Apps, you can use the [WebRequestEventInterface](https://developer.chrome.com/docs/extensions/reference/webviewTag/#type-WebRequestEventInterface) to also modify requests and validate that they are allowed to navigate to your allowed list of sites.
```

Comment:
```suggestion
If your app requires a WebView, make sure that the content within your WebView is not able to escape your application and navigate to content that you do not own or control. This can be done by modifying the [WebView client to override loading certain URLs via an allowlist](<https://developer.android.com/reference/android/webkit/WebViewClient#shouldOverrideUrlLoading(android.webkit.WebView,%20android.webkit.WebResourceRequest)>) on Android or by manually parsing through your site and validating that the links do not link out to third party content that you cannot control as this may enable your users to go to inappropriate content. For Chrome Apps, you can use the [WebRequestEventInterface](https://developer.chrome.com/docs/extensions/reference/webviewTag/#type-WebRequestEventInterface) to also modify requests and validate that they are allowed to navigate to your allowed list of sites.
```